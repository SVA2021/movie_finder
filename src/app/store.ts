import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import authReducer from '../features/auth/authSlice';
import movieFinderReducer from '../features/movieFinder/movieFinderSlice';

export const store = configureStore({
  reducer: {
    auth: authReducer,
		movieFinder: movieFinderReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
