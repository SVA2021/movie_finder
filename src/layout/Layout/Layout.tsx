import {Alert, Backdrop, CircularProgress, Container, Snackbar} from "@mui/material";
import {FC, useState} from "react";
import {Outlet} from "react-router-dom";
import {useAppDispatch, useAppSelector} from "../../app/hooks";
import {Banner} from "../../components";
import {selectError, selectStatus, setError} from "../../features/movieFinder/movieFinderSlice";
import {Footer, Header} from "../index";

export const Layout: FC = () => {

  const dispatch = useAppDispatch();
  const SERVER_STATUS = useAppSelector(selectStatus);
  const SERVER_ERROR = useAppSelector(selectError);
  const [isOpenBanner, setIsOpenBanner] = useState(true);

  const handleCloseError = () => dispatch(setError(null));
  const handleCloseBanner = () => setIsOpenBanner(false);

  return (
    <Container maxWidth='xl' sx={{minHeight: '100vh'}} data-testid={'layout'}>
      <Backdrop open={SERVER_STATUS === 'loading'} sx={{zIndex: 1000}} >
        <CircularProgress color="warning" />
      </Backdrop>
      <Header />
      {isOpenBanner && <Banner closeHandler={handleCloseBanner} />}
      <Outlet />
      <Footer />
      <Snackbar open={!!SERVER_ERROR} autoHideDuration={2500} onClose={handleCloseError}>
        <Alert onClose={handleCloseError} severity="error" sx={{width: '100%'}}>
          {SERVER_ERROR}
        </Alert>
      </Snackbar>
    </Container>
  )
};