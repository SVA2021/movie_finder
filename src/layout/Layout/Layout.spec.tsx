import {render, screen} from "@testing-library/react";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import {store} from "../../app/store";
import {Layout} from "./Layout";

describe('render Layout', () => {

  const setUp = () => render(
    <Provider store={store}>
      <BrowserRouter>
        <Layout />
      </BrowserRouter>
    </Provider>
  );

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('layout')).toMatchSnapshot();
  });

})