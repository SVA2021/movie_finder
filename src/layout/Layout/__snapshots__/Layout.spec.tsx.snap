// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Layout snapshot test 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthXl css-dwngqs-MuiContainer-root"
  data-testid="layout"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root css-1fvlqf9-MuiBackdrop-root"
    style="opacity: 0; visibility: hidden;"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorWarning css-1rnp7nk-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
  <div
    class="MuiBox-root css-7xotxy"
    data-testid="header"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-m1z56k-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-13qhygc-MuiToolbar-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-9cyib4-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-1 css-10f59y6-MuiGrid-root"
          >
            <button
              aria-haspopup="true"
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1qbbts5-MuiButtonBase-root-MuiButton-root"
              id="fade-button"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                data-testid="MenuIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 MuiGrid-grid-md-5 MuiGrid-grid-lg-4 css-1d62rrc-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-h1 css-lswgau-MuiTypography-root"
              data-testid="logo"
            >
              Movie
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                data-testid="LocalMoviesIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 3v2h-2V3H8v2H6V3H4v18h2v-2h2v2h8v-2h2v2h2V3h-2zM8 17H6v-2h2v2zm0-4H6v-2h2v2zm0-4H6V7h2v2zm10 8h-2v-2h2v2zm0-4h-2v-2h2v2zm0-4h-2V7h2v2z"
                />
              </svg>
              Finder
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-1 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4 css-1hub91h-MuiGrid-root"
          >
            <form
              class="MuiBox-root css-hpgf8j"
              data-testid="search-field"
            >
              <div
                class="MuiInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-adornedEnd css-1qmitm0-MuiInputBase-root-MuiInput-root"
              >
                <input
                  class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  placeholder="Movies, series, keywords..."
                  type="text"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1wgghly-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="submit"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="SearchIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </form>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-3 MuiGrid-grid-lg-2 css-3lf7b5-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-h3 css-14at1vf-MuiTypography-root"
              data-testid="styled-text-item"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AccountCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
              Гость
            </p>
          </div>
        </div>
      </div>
    </header>
  </div>
  <a
    class="MuiBox-root css-pw2lgz"
    data-testid="banner"
    href="https://portfolio-sva2021.vercel.app"
    rel="noopener noreferref"
    target="_blank"
  >
    <span
      class="MuiTypography-root MuiTypography-caption css-80zbff-MuiTypography-root"
    >
      s
    </span>
    <span
      class="MuiTypography-root MuiTypography-caption css-80zbff-MuiTypography-root"
    >
      v
    </span>
    <span
      class="MuiTypography-root MuiTypography-caption css-80zbff-MuiTypography-root"
    >
      a
    </span>
    <div
      class="MuiBox-root css-bymsth"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1nsf0ti-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </a>
  <div
    class="MuiBox-root css-1xpmf4r"
    data-testid="footer"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-17dcqu8-MuiTypography-root-MuiLink-root"
          data-testid="styled-link"
          href="https://kinopoiskapiunofficial.tech/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-137b474-MuiTypography-root"
          >
            Backend API
          </p>
        </a>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-17dcqu8-MuiTypography-root-MuiLink-root"
          data-testid="styled-link"
          href="https://portfolio-sva2021.vercel.app/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-137b474-MuiTypography-root"
          >
            Портфолио SVA
          </p>
        </a>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-17dcqu8-MuiTypography-root-MuiLink-root"
          data-testid="styled-link"
          href="https://github.com/SVA2021"
          rel="noopener noreferrer"
          target="_blank"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-137b474-MuiTypography-root"
          >
            Github SVA
          </p>
        </a>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-17dcqu8-MuiTypography-root-MuiLink-root"
          data-testid="styled-link"
          href="https://github.com/SVA2021/movie_finder"
          rel="noopener noreferrer"
          target="_blank"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-137b474-MuiTypography-root"
          >
            Code
          </p>
        </a>
      </div>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1o4n9zm-MuiTypography-root"
    >
      SVA 2023
    </p>
  </div>
</div>
`;
